File: 0986-interval-list-intersections/0986-interval-list-intersections.py
class Solution:
    def intervalIntersection(self, firstList: list[list[int]], secondList: list[list[int]]) -> list[list[int]]:
        n = len(firstList)
        m = len(secondList)
        ans =[]
        first =0
        second =0

        while first < n and second < m:
            if firstList[first][0]<=secondList[second][1] and firstList[first][1]>=secondList[second][0]:
                temp=[0]*2
                temp[0]= max(firstList[first][0],secondList[second][0])
                temp[1]= min(firstList[first][1],secondList[second][1])
                ans.append(temp)
            if firstList[first][1]<secondList[second][1]:
                first+=1
            else:
                second+=1
        return ans

File: 0986-interval-list-intersections/0986-interval-list-intersections.cpp
class Solution {
public:
    vector<vector<int>> intervalIntersection(vector<vector<int>>& firstList, vector<vector<int>>& secondList) {
        int n = firstList.size();
        int m = secondList.size();
        vector<vector<int>> ans;

        int first =0;
        int second =0;

        while(first<n && second<m){
            if(firstList[first][1]>=secondList[second][0] && firstList[first][0]<= secondList[second][1]){
                vector<int> temp(2);
                temp[0]= max(firstList[first][0],secondList[second][0]);
                temp[1]= min(firstList[first][1],secondList[second][1]);
                ans.push_back(temp);
            }
            if(firstList[first][1]<secondList[second][1]){
                first++;
            }
            else{
                second++;
            }
        }
        return ans;
    }
};